networkx/classes/backends.py

```diff
@@ -128,7 +128,10 @@ def _dispatch(func=None, *, name=None):
 
     @functools.wraps(func)
     def wrapper(*args, **kwds):
-        graph = args[0]
+        if args:
+            graph = args[0]
+        else:
+            graph = kwds["G"]  # raise KeyError if not present
```

Comment:
I think it would be better to raise `TypeError` with a message like `f"{name}() missing required positional argument: 'G'"` to better match typical errors and messages when incorrectly calling a function.